CppCoreGuidelines.md

```diff
@@ -19297,26 +19297,38 @@ The [standard](http://eel.is/c++draft/cpp.include) provides flexibility for comp
 the two forms of `#include` selected using the angle (`<>`) or quoted (`""`) syntax. Vendors take
 advantage of this and use different search algorithms and methods for specifying the include path.
 
-Nevertheless, the guidance is to use the quoted form for including files that exist at a relative path to the file containing the `#include` statement (from within the same component or project) and to use the angle bracket form everywhere else, where possible. This encourages being clear about the locality of the file relative to files that include it, or scenarios where the different search algorithm is required. It makes it easy to understand at a glance whether a header is being included from a local relative file versus a standard library header or a header from the alternate search path (e.g. a header from another library or a common set of includes).
+Nevertheless, the guidance is to use the quoted form for including files that exist at a relative path
+to the file containing the `#include` statement and to use
+the angle bracket form everywhere else, where possible. This encourages being clear about the locality
+of the file relative to files that include it, or scenarios where the different search algorithm is
+required. It makes it easy to understand at a glance whether a header is being included from a locally
+relative file, versus a standard library header or a header from the header search path (e.g. a header
+from another library or from a common set of includes).
 
 ##### Example
 
     // foo.cpp:
     #include <string>                // From the standard library, requires the <> form
-    #include <some_library/common.h> // A file that is not locally relative, included from another library; use the <> form
-    #include "foo.h"                 // A file locally relative to foo.cpp in the same project, use the "" form
-    #include "foo_utils/utils.h"     // A file locally relative to foo.cpp in the same project, use the "" form
-    #include <component_b/bar.h>     // A file in the same project located via a search path, use the <> form
+    #include <some_library/common.h> // A file that is not locally relative, included from another library: use the <> form
+    #include "foo.h"                 // A file located relative to foo.cpp: use the "" form
+    #include "foo_utils/utils.h"     // A file located relative to foo.cpp: use the "" form
+    #include <component_b/bar.h>     // A file located via the header search path: use the <> form
```

Comment:
Users commonly believe that if an include is from the same library (or codebase or project or whatever you want to call it), then it must use the `""` form. However, could still have been found via the search path so that belief is wrong.

If there's one thing I'm trying to do with all of these comments, it's to fix this misconception. I fear that this PR pulls us back towards reinforcing that misconception. It's not quite clear enough that _component_b_ is part of the same whatchamacallit.

Reply:
I struggled with this too; the whatchamacallit is, well... a project. which means we have to choose between defining what a project is (pro: being more crisp about how to include things from a project) or leaving it intentionally up to the user to define (pro: steering clear of defining what a project is in the guidelines, which is by nature a very liberal concept).

I think I'm ok erring on the side of _not_ defining what a project is here, and letting users figure out how best to include things from within one. The guideline is clear about how "" should be used, but if for some reason they're still adamant that "" must be used within their project, they'll have to align that with their policy about _relatively_ accessing everything within their project (this should be intentionally left up to the user: some are ok with "..\bar\bar.h", some might want <foo/bar/bar.h>)

As a funny anecdote, I'm actually trying to dispel the _opposite_ myth in my codebase: a lot of developers are under the impression that <> must always be used from within a project (from what I can tell it's a myth created because so many have been bitten by differences in the "" vs <> they inevitably decide to stick with one)

Reply:
> a lot of developers are under the impression that <> must always be used from within a project (from what I can tell it's a myth created because so many have been bitten by differences in the "" vs <> they inevitably decide to stick with one)

That pretty much sums up my attitude, except: `s/must always/should always/` and `s/myth/guideline/`. 🙂

"A lot of cats are under the impression that stoves are not for sitting on (from what I can tell it's a myth created by once sitting on a hot stove)" 😉

Reply:
"project" is mostly a Microsoft IDE-ism.  We need to look beyond the particular workflow of a specific environment and implementation at one point of software organization.